Automattic/jetpack
# Licensing: Move licensing endpoints to package

In preparation of moving licensing into the `my-jetpack` package, this pull request moves the licensing endpoints out of the Jetpack plugin and into the `licensing` package. This will then allow any consumer of the `licensing` package to use the endpoints without requiring the Jetpack plugin itself.

#### Changes proposed in this Pull Request:

* Moves licensing endpoints out of Jetpack plugin into licensing package.

#### Jetpack product discussion



NA

#### Does this pull request change what data or activity we track or use?



No

#### Testing instructions:






Newer licenses flow:

- Checkout PR
- Set `define( 'JETPACK_AUTOLOAD_DEV', true );`  constant (I think 😄 )
- Go to https://cloud.jetpack.com/pricing and purchase a product
- On next page, scroll to end of site selector and select manual setup
- Click through the next step
- Copy license
- Go to Jetpack dashboard and go to "My Plan" section
- Click "Activate Product" button
- Enter in a key that will fail, such as `this_fails`
- Verify the key fails as expected and that a successful API response comes back
- Enter in the copied key
- Verify that there are several licensing calls that succeed
- Go to "My Plan" section again and verify that your plan/product shows up

Older partnership license flow:

- Set `add_filter( 'jetpack_licensing_ui_enabled', '__return_true' );`
- Purchase another key
- Go to "My Plan" section of dashboard
- Enter license key in the input that shows below the list of any existing plans or products
- Verify a successful response
- Wait a few seconds and reload the page
- Verify subscription exists in the list of plans/products

Note: The `licensing/user/licenses` doesn't appear to actually be used in the plugin? It is used in an action but I don't see that action being used anywhere.

<img width="1559" alt="Screen Shot 2022-03-30 at 3 09 23 PM" src="https://user-images.githubusercontent.com/1126811/160924729-d4cc5e38-fd56-45a2-8ee1-4e49ccd864ff.png">

<img width="1559" alt="Screen Shot 2022-03-30 at 3 09 51 PM" src="https://user-images.githubusercontent.com/1126811/160924750-fdff24b0-8b31-4d8a-8d9d-b38bb0ecbc79.png">

<img width="1559" alt="Screen Shot 2022-03-30 at 3 54 47 PM" src="https://user-images.githubusercontent.com/1126811/160929748-805fb802-a0ca-4408-b9f1-5c49a0736a16.png">
